src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentState.cs

```diff
@@ -107,6 +114,12 @@ protected static ValueSource<TextAndVersion> CreateRecoverableText(TextLoader lo
             }
             catch (IOException e)
             {
+                if (++retries <= MaxRetries)
```

Comment:
FileNotFoundException is-a IOException. Should it be treated differently (no retries)?I don't really want to wait 5 seconds for that.